lib/src/prayer_times.dart

```diff
@@ -227,22 +229,22 @@ class PrayerTimes {
     tempMaghrib = sunsetComponents;
     if (calculationParameters.maghribAngle != null) {
       final angleBasedMaghrib = TimeComponents.fromDouble(solarTime.hourAngle(
-              -1 * calculationParameters.maghribAngle, true))
+              -1 * calculationParameters.maghribAngle!, true))!
           .dateComponents(date);
-      if (sunsetComponents.isBefore(angleBasedMaghrib) &&
-          tempIsha.isAfter(angleBasedMaghrib)) {
+      if (sunsetComponents!.isBefore(angleBasedMaghrib) &&
+          tempIsha!.isAfter(angleBasedMaghrib)) {
         tempMaghrib = angleBasedMaghrib;
       }
     }
 
-    if (error || tempAsr == null) {
-      // if we don't have all prayer times then initialization failed
-      _fajr = null;
-      _sunrise = null;
-      _dhuhr = null;
-      _asr = null;
-      _maghrib = null;
-      _isha = null;
+    if (error ||
+        tempAsr == null ||
+        tempFajr == null ||
+        tempSunrise == null ||
+        tempDhuhr == null ||
+        tempIsha == null ||
+        tempMaghrib == null) {
+      throw ArgumentError('Invalid Coordinates');
```

Comment:
I think throwing ArgumentError('Invalid Coordinates') here isn't appropriate. We should throw Invalid Coordinates in the Coordinates class.

Reply:
Here we need a more meaningful and descriptive message with a RuntimeError, the user needs to know why failed to initialize.